src/awkward/_slicing.py

```diff
@@ -266,8 +266,7 @@ def normalise_item_nested(item):
             ak.contents.UnmaskedArray,
         ),
     ):
-        next = item.simplify_optiontype()
-        return normalise_item_nested(next)
+        return normalise_item_nested(item)
```

Comment:
This is one of the few that doesn't replace a `simplify_optiontype` with a `simplified` classmethod—not a pure refactor—so it's possible that non-canonical layouts pass through here.

However, I think this was sanitizing user input, and after constructors forbid non-canonical structures, user input can't be insane. So while this opens the door to error, the next PR will shut it.